### PR TITLE
Remove redirect-based trigger-registration API surface

### PR DESCRIPTION
**Pull request by apasel422**
And update it to use JSON parsing.

#386

***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/conversion-measurement-api/pull/451.html" title="Last updated on May 26, 2022, 6:29 PM UTC (a43f3ef)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/451/d2e33b2...apasel422:a43f3ef.html" title="Last updated on May 26, 2022, 6:29 PM UTC (a43f3ef)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -340,12 +334,13 @@ An event-level trigger configuration is a [=struct=] with the following items:
 :: A 64-bit integer.
 : <dfn>filters</dfn>
 :: An [=attribution filter map=].
+: <dfn>negated filters</dfn>
```
Comment (johnivdel) [inline index.bs:337]: I believe this is being addressed on https://github.com/WICG/attribution-reporting-api/pull/435, but both PRs look ccompatible.

File: index.bs [hunk 2]
```diff
@@ -787,87 +782,92 @@ Issue: Should fake reports respect the user agent's [=max reports per attributio
...
+To <dfn noexport>parse trigger-registration JSON</dfn> given a [=string=]
+|json|, a [=site=] |destination|, and an [=origin=] |reportingOrigin|:
+
+1. Assert: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |value| be the result of running
+    [=parse a JSON string to an Infra value=] with |json|.
+1. If |value| is not an [=ordered map=], return null.
+1. Let |eventTriggers| be the result of running [=parse event triggers=]
+    with |value|.
+1. If |eventTriggers| is null, return null.
```
Comment (johnivdel) [inline index.bs:838]: This is assuming https://github.com/WICG/attribution-reporting-api/pull/448 is landed, but will also need to change when we add aggregate parsing, correct?

File: index.bs [hunk 3]
```diff
@@ -787,87 +782,92 @@ Issue: Should fake reports respect the user agent's [=max reports per attributio
...
+1. Let |debugKey| be null.
+1. If |value|["`debug_key`"] [=map/exists=] and is a [=string=]:
+    1. Set |debugKey| to the result of applying the
+        <a spec="html">rules for parsing non-negative integers</a> to
+        |value|["`debug_key`"].
+    1. If |debugKey| is an error, set |debugKey| to null.
+1. Let |aggregatableTrigger| be an [=attribution aggregatable trigger=] with
```
Comment (johnivdel) [inline index.bs:845]: unrelated:

Does this nested structure give us much? Should we just include both of these lists directly on trigger?

**Review by johnivdel — COMMENTED** [review]
generally lgtm

File: index.bs [hunk 4]
```diff
@@ -728,58 +723,93 @@ that controls how many [=attribution sources=] can be in the
...
-1. Let |eventConfig| be a new [=event-level trigger configuration=] with the items:
-    : [=event-level trigger configuration/trigger data=]
-    :: |eventSourceTriggerData|
-    : [=event-level trigger configuration/dedup key=]
-    :: |dedupKey|
-    : [=event-level trigger configuration/priority=]
-    :: |triggerPriority|
-    : [=event-level trigger configuration/filters=]
-    :: «[ "`source_type`" → « "`event`" » ]»
+To <dfn>parse event triggers</dfn> given an [=ordered map=] |map|:
```
[the remark this answers is not in the document]
Comment (apasel422) [reply index.bs:785]: The map is the result of https://infra.spec.whatwg.org/#parse-a-json-string-to-an-infra-value so the keys have to be strings, yes.

File: index.bs (hunk 2 above)
Comment (apasel422) [reply index.bs:838]: Yes.

File: index.bs (hunk 3 above)
Comment (apasel422) [reply index.bs:845]: No, it doesn't get us much. Will remove in a separate PR.

**Review by johnivdel — APPROVED** [review]

**Pull request merged**